lib/Backends/OpenCL/OpenCL.cpp

```diff
@@ -619,11 +622,8 @@ static void topK(Tensor &outW, Tensor &indW, Tensor &inW, size_t k) {
   }
 }
 void OpenCLFunction::execute(Context &ctx) {
-  auto copiedToDeviceBytes = copyMutableWeightsToDevice();
-  (void)copiedToDeviceBytes;
-  DEBUG_GLOW(llvm::dbgs() << "Copied " << copiedToDeviceBytes
-                          << " bytes to OpenCL device\n");
-
+  copyConstantsToDevice();
```

Comment:
It seems like it would copy the constant to the device every time, which is not good. Also, IIRC, you cannot run simultaneously two instances of `OpenCLFunction::execute` for the same IRFunction. Thus, I'd suggest allocating the memory on first call of the `execute` and reusing it afterwards. 

With this approach, you can copy the constants only once, which improves the performance a lot.